cf-agent/verify_users_pam.c

```diff
@@ -668,6 +720,19 @@ static bool ChangePassword(const char *puser, const char *password, PasswordForm
     return successful;
 }
 
+static bool IsHashLocked(const char *hash)
+{
+#ifdef HAVE_PW
+    /* Accounts are locked by prepending "*LOCKED*" to the password
+     * hash on FreeBSD. */
```

Comment:
Comment indicates that this is specific to FreeBSD, but macro is `HAVE_PW` which one is correct?

Reply:
> Regarding comment [1], the user management tool on FreeBSD is pw. The way pw locks user accounts is by prepending the password hash with LOCKED. This way the password hash does not match your hashed password, thus you are unable to login using your password. The string used to prepend the password hash can be arbitrary. If you stick to LOCKED, you will be able to unlock a user account with pw. The operating system, FreeBSD in particular, does not care about what string you use. This mechanism of altering the password hash of the user entry can also be applied to any other system. I can change the comment and replace FreeBSD with pw as there are a few other operating systems forked from FreeBSD. Some of the prominent ones are DragonflyBSD, FreeSBIE, FreeNAS, pfSense and m0n0wall.

> Would that change be OK?

@xaxo05 It's easier if you reply inside this thread, instead of adding more top level comments.

Regarding the change, something like this would be sufficient:

```
/* Accounts are locked by prepending "*LOCKED*" to the password
     * hash on FreeBSD and other systems using pw. */
```

It's just confusing when the macro and the comment seem to contradict each other. (Specific to pw or specific to freebsd).